### PR TITLE
python mode / lsp : Added support for Microsoft Python Language Server

### PR DESCRIPTION
**Pull request by cormacc**
Added support for MS Python Language Server as alternate backend.
Options are now `anaconda`, `lsp` or `lsp-ms`.

See layer README for configuration info.

### CODE REVIEW

File: layers/+lang/python/packages.el [hunk 1]
```diff
@@ -407,3 +408,20 @@ fix this issue."
...
+    :if (eq python-backend 'lsp-ms)
+    :ensure nil
+    :config
+
+    (if python-lsp-git-root
+      ;; Use dev version of language server checked out from github
+      (progn
+        (setq lsp-python-ms-dir
+          (expand-file-name (concat python-lsp-git-root "/output/bin/Release/")))
+        (message ( concat "lsp-python-ms: Using version at " lsp-python-ms-dir)))
```
Comment (yyoncho) [inline layers/+lang/python/packages.el]: more common is to use %s with message instead of concat.

Comment (cormacc) [reply layers/+lang/python/packages.el]: Thanks @yyoncho -- amended/pushed

**Comment by cormacc** [comment]
I should get a chance to address the feedback tomorrow at some stage. @robbyoconnor re. antipattern/specifying language server, do you mean to revert to `anaconda` and `lsp` as allowed values for `backend`, and add another config variable `language-server`, values `default`, and 'microsoft' (or `nil` and `microsoft`)?

**Comment by yyoncho** [comment]
> values `default`, and 'microsoft' (or `nil` and `microsoft`)?

IMO better names are the lsp-mode server ids. And just a heads up - at some point, we(lsp-mode team) will provide and option to manage the server priorities using variables so ideally, this configuration should disappear.

**Comment by robbyoconnor** [comment]
@cormacc what I'm saying is specify the language server to use, not provide a separate backend if we want to change the language server that we use. So yes, I am saying to keep `anaconda` and `lsp` and specify the language server to use.

**Comment by cormacc** [comment]
Incorporated review feedback and rebased

File: CHANGELOG.develop [hunk 2]
```diff
@@ -2030,7 +2030,10 @@ Other:
 - Use =python-mode= for SCons script files (thanks to shanemikel)
 - Added LSP support, which can be used by enabling the =lsp= layer and setting
   the =python= layer's =python-backend= variable to =lsp=
-  (thanks to Yuan Fu and Sylvain Benner)
+  (thanks to Yuan Fu and Sylvain Benner). The LSP backend can use either the
```
Comment (duianto) [inline CHANGELOG.develop]: Fill issue. Start on a newline below the previous thanks parentheses. With a leading `-` (dash).

File: layers/+lang/python/packages.el [hunk 3]
```diff
@@ -420,3 +422,20 @@ fix this issue."
                  (eq 'yapf python-formatter))
         (add-hook 'python-mode-hook 'yapf-mode)))
     :config (spacemacs|hide-lighter yapf-mode)))
+
+;;https://github.com/andrew-christianson/lsp-python-ms
+;;Not on MELPA
```
Comment (duianto) [inline layers/+lang/python/packages.el]: Should this comment be removed?
@yyoncho Mentioned in a previous comment that it's on melpa now.

Comment (cormacc) [reply layers/+lang/python/packages.el]: Yep - had updated the layer to use the melpa version, but forgotten to strip the comment. Thanks

File: CHANGELOG.develop (hunk 2 above)
Comment (cormacc) [reply CHANGELOG.develop]: Had been trying to keep the python LSP stuff in a single changelog entry, but split now.

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.

**Pull request merged**